ledger-core/virtual/object/working_table.go

```diff
@@ -14,38 +14,38 @@ import (
 )
 
 type WorkingTable struct {
-	requests [3]*WorkingList
+	requests [contract.InterferenceFlagCount - 1]*WorkingList
 	results  map[reference.Global]Summary
 }
 
 func NewWorkingTable() WorkingTable {
 	var rt WorkingTable
 
-	for i := 1; i < contract.InterferenceFlagCount; i++ {
-		rt.requests[i] = NewWorkingList()
+	for i := len(rt.requests) - 1; i >= 0; i-- {
+		rt.requests[i] = newWorkingList()
 	}
 
 	rt.results = make(map[reference.Global]Summary)
 
 	return rt
 }
 
-func (wt *WorkingTable) GetList(flag contract.InterferenceFlag) *WorkingList {
-	if flag.IsZero() || flag >= contract.InterferenceFlagCount {
-		panic(throw.IllegalValue())
+func (wt WorkingTable) GetList(flag contract.InterferenceFlag) *WorkingList {
+	if flag > 0 && int(flag) <= len(wt.requests) {
+		return wt.requests[flag - 1]
 	}
-	return wt.requests[flag]
+	panic(throw.IllegalValue())
 }
 
-func (wt *WorkingTable) GetResults() map[reference.Global]Summary {
+func (wt WorkingTable) GetResults() map[reference.Global]Summary {
 	return wt.results
 }
 
-func (wt *WorkingTable) Add(flag contract.InterferenceFlag, ref reference.Global) bool {
+func (wt WorkingTable) Add(flag contract.InterferenceFlag, ref reference.Global) bool {
 	return wt.GetList(flag).add(ref)
 }
 
-func (wt *WorkingTable) SetActive(flag contract.InterferenceFlag, ref reference.Global) bool {
+func (wt WorkingTable) SetActive(flag contract.InterferenceFlag, ref reference.Global) bool {
```

Comment:
this is not really good, because it relies on map shallow copy mechanics. If the object is changed by method, it's better to actually have ptr receiver